### PR TITLE
Update documentation and tests to [docsrs]

### LINKED ISSUE #2423: TUTORIAL.md: Replace Wercker with some public service API
:question: **Question**

Attempt to find out more info about Wercker leads to Oracle page about company acquisition. I think that's not as interesting and inspiring as working with some open API, such as Travis CI, GitLab or Travis. Would be nice to see some open source examples there.

### PR DESCRIPTION
**Pull request by PyvesB**
Updates the main tutorial and the service-tests one to use our Docs.rs service instead of the no longer existing Wercker service. It's a nice an simple service with everything publicly available. Closes #2423.

One of the error behaviours changed on the upstream API, I've also updated the implementation and tests to reflect that.

### CODE REVIEW

File: doc/service-tests.md [hunk 1]
```diff
@@ -115,80 +119,100 @@ That's looking good!
...
 
-Having covered the typical and custom cases, we'll move on to errors. We should include a test for the 'not found' response and also tests for any other custom error handling. The Wercker integration defines a custom error condition for 401 as well as a custom 404 message:
+We can also add a test for a unsuccessful documentation build:
 
-```js
-httpErrors: {
-  401: 'private application not supported',
-  404: 'application not found',
-}
+```j
```
Comment (chris48s) [inline doc/service-tests.md:161]:
```suggestion
```js
```

File: doc/service-tests.md [hunk 2]
```diff
@@ -115,80 +119,100 @@ That's looking good!
...
+t.create('Failing docs for version')
+  .get('/tensorflow/0.16.1.json')
```
Comment (chris48s) [inline doc/service-tests.md:163]: There's a couple of places here where we are switching between using tensorflow and tokio. Can we either stick to one, or explain more explicitly why we are using different packages in some examples.

**Review by chris48s — COMMENTED** [review]
Couple of minor points to follow up, but I'm on-board with this. Good to update these docs to use a more relevant example.

Also closes a long-standing issue https://github.com/badges/shields/issues/2423 :D

File: doc/service-tests.md (hunk 2 above)
Comment (PyvesB) [reply doc/service-tests.md]: There you go, tokio also has a couple of versions with failing builds.

**Review by chris48s — APPROVED** [review]

**Pull request merged**